File: python3/remove_element.py
def removeElement(nums, val):
    j = 0
    for i in range(len(nums)):
        if nums[i] != val:
            nums[j] = nums[i]
            j += 1
    return j


nums = [3, 2, 2, 3]
length = removeElement(nums, 3)
print('ans:', nums[:length])

File: cpp/remove_element.cpp
#include<iostream>
#include<vector>
using namespace std;

int removeElement(vector<int>& nums, int val) {
    int j = 0;
    for(int i=0; i<nums.size(); i++){
        if(nums[i] != val){
            nums[j] = nums[i];
            j++;
        }
    }
    return j;
}

int main(){
    vector<int> nums{3, 2, 2, 3};
    int length = removeElement(nums, 3);
    for(int i=0; i<length; i++){
        cout<<nums[i]<<endl;
    }
}
